src/plugin/Manager.cc

```diff
@@ -195,6 +195,23 @@ bool Manager::ActivateDynamicPluginInternal(const std::string& name, bool ok_if_
 
 	DBG_LOG(DBG_PLUGINS, "Activating plugin %s", name.c_str());
 
+	// If there's a plugin with the same name already, report an error an let
+	// the user do the conflict resolution.
+	for ( const auto& p : *all_plugins )
+		{
+		auto lower_name = util::strtolower(name);
```

Comment:
Move this above the loop to avoid doing it repeatedly.

Reply:
Thanks, done. I was mostly concerned about the O(n^2) this ends up being, but there aren't that many plugins :crossed_fingers: and if it gets too slow we can store plugins in a lookup table I guess.